pmd-java/src/test/java/net/sourceforge/pmd/lang/java/symbols/internal/SymImplementation.java

```diff
@@ -145,4 +162,40 @@ private void assertParameterMatch(Parameter p, JFormalParamSymbol pSym) {
         assertEquals(expectedType, pSym.getTypeMirror(Substitution.EMPTY));
     }
 
+    /**
+     * In order to test simultaneously types defined in the same compilation unit,
+     * we must store them somewhere. The fixture stores the parsed AST and allows
+     * convenient access to the parsed types. This makes sure that we don't parse
+     * the file once per lookup.
+     */
```

Comment:
Yes but the tests that use those test a single type anyway, so it doesn't matter for them. It's useful in the new tests I added though, because several types are defined in the same source file, and we want to test the connections between those symbols.